File: excel-sheet-column-number.rb
# @param {String} s
# @return {Integer}
def title_to_number(s)
  sum = 0
  a = 1
  s.chars.reverse.each do |c|
    sum += (c.ord - 'A'.ord + 1) * a
    a *= 26
  end
  return sum
end

File: excel-sheet-column-number.py
class Solution:
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = 0
        power = 1
        for c in s[::-1]:
            ans += power * (ord(c) - ord("A") + 1)
            power *= 26
        return ans
